sklearn/model_selection/_split.py

```diff
@@ -2495,6 +2497,162 @@ def _build_repr(self):
     return "%s(%s)" % (class_name, _pprint(params, offset=len(class_name)))
 
 
+class GroupTimeSeriesSplit(_BaseKFold):
+    """Time Series cross-validator variant with non-overlapping groups.
+
+    Provides train/test indices to split time series data samples
+    that are observed at fixed time intervals according to a
+    third-party provided group.
+    In each split, test indices must be higher than before, and thus shuffling
+    in cross validator is inappropriate.
+
+    This cross-validation object is a variation of :class:`KFold`.
+    In the kth split, it returns first k folds as train set and the
+    (k+1)th fold as test set.
+
+    The same group will not appear in two different folds (the number of
+    distinct groups has to be at least equal to the number of folds).
+
+    Note that unlike standard cross-validation methods, successive
+    training sets are supersets of those that come before them.
+
+    The groups should be continuous. For Example:
+    np.array(['a', 'a', 'a', 'a', 'a', 'a',\
+                           'b', 'b', 'b', 'b', 'b',\
+                           'c', 'c', 'c', 'c',\
+                           'd', 'd', 'd'])
+
+    Non-continuous groups like below will give an error.
+    np.array(['a', 'a', 'a', 'a', 'a', 'a',\
+                           'b', 'b', 'b', 'b', 'b',\
+                           'a', 'c', 'c', 'c',\
+                           'b', 'd', 'd'])
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_splits : int, default=5
+        Number of splits. Must be at least 2.
+
+    max_train_size : int, default=None
+        Maximum size for a single training set.
```

Comment:
Please specify the units: this size is the number of groups, isn't it?

Reply:
done